crypto/rsa/rsa_ossl.c

```diff
@@ -705,6 +735,47 @@ static int rsa_ossl_mod_exp(BIGNUM *r0, const BIGNUM *I, RSA *rsa, BN_CTX *ctx)
         BN_free(dmp1);
     }
 
+    /* calculate m_i in multi-prime case */
```

Comment:
I see no reason for why m_i can't be calculated in second loop without using m[].

Reply:
Well, it might make sense to calculate them separately if one aims for vectorization... I suppose this comment can be disregarded...

Reply:
However, in the context, I'd like to argue in favour of cleaning up this section by

- removing cc is re-using c from previous block (preferably renaming it as cI to denote that fact that it's "copy" of I);
- removing dmq1 and dmp1 in previous block and using di throughout;

Reply:
No real impacts on the functionality, so I decide not to squash the two loops...

Reply:
Was it reply to first comment or 3rd? In either case, computational outcome is not sole goal. Code has to be readable, and it's not inappropriate if attempt is made to minimize resource utilization.

As for 2nd comment. After some extra consideration I'd say that we don't have to take into account vectorization. Not because that it's off the table, but because it's more than likely to be solved as dedicated code paths anyway. I mean generic code path can and probably should calculate m_i as it goes, i.e in one loop.

Reply:
Replied to the first comment. But I still would like to leave a comment in the code and change that part in the future...